core/src/main/java/org/mule/runtime/core/connector/DefaultReplyToHandler.java

```diff
@@ -62,12 +60,12 @@ public void processReplyTo(final MuleEvent event, final MuleMessage returnMessag
 
         // make sure remove the replyTo property as not cause a a forever
         // replyto loop
-        returnMessage.removeProperty(MuleProperties.MULE_REPLY_TO_PROPERTY, OUTBOUND);
+        returnMessage.removeOutboundProperty(MuleProperties.MULE_REPLY_TO_PROPERTY);
```

Comment:
But set... will remove if null is passed. Also, by just removing the property the new attribute that was included is not being cleaned.